apps/shell/bin/setup

```diff
@@ -1,14 +1,57 @@
 #!/usr/bin/env ruby
 require "pathname"
 require "rake"
+require "yaml"
 
 include FileUtils
 
 # path to your application root.
 APP_ROOT = Pathname.new File.expand_path("../../",  __FILE__)
 
-# application settings
-DEFAULT_SSHHOST = ENV["DEFAULT_SSHHOST"]
+# Set Whitelist env variable
+whitelist = []
+if ENV['SSHHOST_WHITELIST']
+  whitelist = ENV['SSHHOST_WHITELIST'].split(':')
+end
+
+# Determine sshhosts from cluster config files
+sshhost = []
+Dir.glob('/etc/ood/config/clusters.d/*.y*ml') do |config_file|
+  file = YAML.load_file(config_file)
+  if file.has_key?('v2')
+    v2 = file['v2']
+    if (v2.has_key?('metadata')) #check for hidden attribute in metadata
+      metadata = v2['metadata']
+      if metadata['hidden'] 
+        next #if true, skip to next config file
+      end
+    end
+
+    if (v2.has_key?('login')) #Skip over file if no login info
+      login = v2['login']
+      host = login['host']
+      default = login['default']
+      if default
+        sshhost.unshift(host)
+      else
+        sshhost << host
+      end
+    end
+  else
+    puts "#{config_file} does not follow config schema v2. See osc.github.io/ood-documentation for more details"
+  end
+end
```

Comment:
The reading of the cluster configs to add to the whitelist should be done in app.js. `bin/setup` is only executed when building the shell app.